Packs/IBMStorageScale/Integrations/IBMStorageScale/IBMStorageScale.py

```diff
@@ -0,0 +1,546 @@
+        cleaned_hashes = dict(sorted_hashes[:MAX_STORED_HASHES])
+
+    # Update last run with cleaned hashes
+    last_run = demisto.getLastRun()
+    last_run["event_hashes"] = cleaned_hashes
+    demisto.setLastRun(last_run)
+    demisto.debug(f"Stored {len(cleaned_hashes)} event hashes for deduplication")
+
+
+def deduplicate_events(events: list[dict[str, Any]]) -> tuple[list[dict[str, Any]], dict[str, int]]:
+    """
+    Remove duplicate events using stored hashes and return deduplication stats.
+
+    Returns:
+        tuple: (deduplicated_events, stats_dict)
+    """
+    if not events:
+        return events, {"total_events": 0, "duplicates_found": 0, "unique_events": 0}
+
+    stored_hashes = get_stored_event_hashes()
+    new_hashes = {}
+    deduplicated_events = []
+    duplicates_found = 0
+
+    for event in events:
+        event_hash = generate_event_hash(event)
+        event_time = event.get("entryTime", datetime.utcnow().isoformat() + "Z")
+
+        # Check if this event hash already exists
+        if event_hash in stored_hashes:
+            duplicates_found += 1
+            demisto.debug(f"Duplicate event found with hash {event_hash[:12]}...")
+            continue
+
+        # Add to deduplicated events and track new hash
+        deduplicated_events.append(event)
+        new_hashes[event_hash] = event_time
+
+    # Merge new hashes with stored hashes and store
+    all_hashes = {**stored_hashes, **new_hashes}
+    store_event_hashes(all_hashes)
+
+    stats = {
+        "total_events": len(events),
+        "duplicates_found": duplicates_found,
+        "unique_events": len(deduplicated_events),
+        "stored_hashes_count": len(all_hashes),
+    }
+
+    demisto.debug(f"Deduplication complete: {stats}")
+    return deduplicated_events, stats
+
+
+def get_fetch_start_time() -> datetime:
+    """
+    Get the start time for the fetch window from the last run object.
+    Returns a datetime object.
+    """
+    last_run = demisto.getLastRun()
+    last_fetch_time_str = last_run.get("last_fetch_time")
+
+    if last_fetch_time_str:
+        demisto.debug(f"Using last fetch time from last run: {last_fetch_time_str}")
+        # Parse ISO 8601 format, handling 'Z' for UTC
+        return datetime.fromisoformat(last_fetch_time_str.replace("Z", "+00:00"))
+    else:
+        # First run - use default lookback period based on DEFAULT_FIRST_FETCH_MINUTES (in days)
+        start_time = datetime.utcnow() - timedelta(days=DEFAULT_FIRST_FETCH_MINUTES)
+        demisto.debug(f"First run - using default lookback time: {start_time.isoformat()}Z")
+        return start_time
+
+
+def update_last_run_time(new_fetch_time: datetime) -> None:
+    """
+    Update the last run object with the given timestamp.
+    """
+    last_run = demisto.getLastRun()
+    # Store in UTC ISO format with 'Z', handling timezone-aware datetime
+    time_str = new_fetch_time.isoformat()
+    if not time_str.endswith("Z"):
+        time_str += "Z"
```

Comment:
perhaps can also be handled in the same helper function to streamline time zone manipulation, as different areas have a different time zone requirement